File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        //We can use 2 pointers here array is sorted already
        map<int,int> mp;
        vector<int> v;
        for(int i=0;i<n;i++){
            if(mp.find(target-nums[i])!=mp.end()){
                v.push_back(mp[target-nums[i]]+1);
                v.push_back(i+1);
            }
            mp[nums[i]]=i;
        }
        return v;
    }
};

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      # alkasnlkas
        dic={}
        for i,num in enumerate(nums):
            if target-num in dic:
                return [dic[target-num]+1,i+1]
            dic[num]=i;
            
        